hnn_core/network.py

```diff
@@ -219,47 +293,443 @@ def __repr__(self):
+        drive['events'] = []
+        self._attach_drive(name, drive, weights_ampa, weights_nmda, location,
+                           space_constant, dispersion_time)
+
+    def add_bursty_drive(self, name, *,
+                         distribution=None, t0=None, sigma_t0=None, T=None,
+                         burst_f=None, burst_sigma_f=None, numspikes=None,
+                         repeats=None, weights_ampa=None, weights_nmda=None,
+                         location=None, dispersion_time=0.1,
+                         space_constant=100., seedcore=42):
+        """Add a bursty (rhythmic) external drive to all cells of the network
+
+        Parameters
+        ----------
+        name : str
+            Unique name for drive
+        distribution : str
+            The distribution for each burst. Either 'normal' or 'uniform'.
+        t0 : float
+            Start time of the burst trains
+        sigma_t0 : float
+            If greater than 0 randomize start times with standard deviation
+            sigma_t0.
+        T : float
+            End of burst trains
+        burst_f : float
+            The frequency of input bursts.
+        burst_sigma_f : float
+            The standard deviation of spike times within burst. Only applied
+            to 'normal' distribution.
+        numspikes : int
+            The events per cycle. Must be 1 or 2. If it is 2, then
+            return doublets 10 ms apart. This is the spikes/burst
+            parameter in GUI.
+        repeats : int
+            The number of repeats.
+        weights_ampa : dict or None
+            Synaptic weights of AMPA receptors on each targeted cell type
+            (keys). Weights can be zero or left omitted.
+        weights_nmda : dict or None
+            Synaptic weights of NMDA receptors on each targeted cell type
+            (keys). Weights can be zero or left omitted.
+        location : str
+            Target location of synapses ('distal' or 'proximal')
+        space_constant : float
+            Describes lateral dispersion (from column origin) of synaptic
+            delays within the simulated column
+        dispersion_time : float or dict
+            Synaptic delay at the column origin, dispersed laterally as a
+            function of the space_constant. If float, applies to all target
+            cell types. Use dict, to create dispersion->cell mapping.
+        seedcore : int
+            Optional initial seed for random number generator (default: 42).
+            Each artificial drive cell has seed = seedcore + gid
+        """
+        if (distribution is None or t0 is None or sigma_t0 is None or
+                T is None or burst_f is None or burst_sigma_f is None or
+                numspikes is None or repeats is None):
+            raise ValueError('distribution, t0, sigma_t0, T, burst_f, '
+                             'burst_sigma_f, numspikes and repeats must all '
+                             'be defined')
```

Comment:
This is a bit weird. See my [comment above](https://github.com/jonescompneurolab/hnn-core/pull/221/files#r549504598)

Reply:
agreed, removed